python_modules/dagster/dagster/_daemon/sensor.py

```diff
@@ -638,8 +639,12 @@ def _evaluate_sensor(
                     asset_selection=stale_assets, stale_assets_only=False
                 )
 
-        target_data: ExternalTargetData = check.not_none(
-            external_sensor.get_target_data(run_request.job_name)
+        target_data: Optional[ExternalTargetData] = external_sensor.get_target_data(
+            run_request.job_name
+        ) or ExternalTargetData(
+            pipeline_name=check.not_none(run_request.job_name),
+            mode=DEFAULT_MODE_NAME,
+            solid_selection=None,
```

Comment:
this is a standout bit of weirdness - especially given `asset_selection` being on `RunRequest`